src/PublicApiGenerator/ApiGenerator.cs

```diff
@@ -675,16 +673,30 @@ static void AddPropertyToTypeDeclaration(CodeTypeDeclaration typeDeclaration, Pr
                 ? new CodeTypeReference(member.PropertyType.Name)
                 : member.PropertyType.CreateCodeTypeReference(member);
 
+            var propertyName = member.Name;
             var property = new CodeMemberProperty
             {
-                Name = member.Name,
+                Name = propertyName,
                 Type = propertyType,
                 Attributes = propertyAttributes,
                 CustomAttributes = CreateCustomAttributes(member, attributeFilter),
                 HasGet = hasGet,
                 HasSet = hasSet
             };
 
+            // DefaultMemberAttribute on type gets propagated to IndexerNameAttribute
+            var defaultMemberAttribute = typeDeclarationInfo.CustomAttributes.SingleOrDefault(x =>
+                x.AttributeType.FullName == "System.Reflection.DefaultMemberAttribute");
+            var defaultMemberAttributeArgument = defaultMemberAttribute?.ConstructorArguments.SingleOrDefault();
```

Comment:
This `SingleOrDefault` makes a hard assumption that no one will ever declare and use a constructor with two parameters on an attribute with the name `System.Reflection.DefaultMemberAttribute`, which is probably true.

Reply:
Unless someone would redeclare the property in that namespace yes.